core/templates/dev/head/pages/topic_editor/questions/QuestionsTabDirective.js

```diff
@@ -118,6 +125,117 @@ oppia.directive('questionsTab', [
             });
           };
 
+          $scope.getSuggestedQuestions = function() {
+            $scope.questionSuggestionThreads = [];
+            $http.get('/generalsuggestionlisthandler', {
+              params: {
+                target_type: 'topic',
+                target_id: $scope.topic.getId(),
+                suggestion_type: 'add_question'
+              }
+            }).then(function(suggestionsResponse) {
+              $http.get(
+                '/threadlisthandlerfortopic/' + $scope.topic.getId()).then(
+                function(threadsResponse) {
+                  var suggestionThreads = (
+                    threadsResponse.data.suggestion_thread_dicts);
+                  var suggestions = suggestionsResponse.data.suggestions;
+                  if (suggestionThreads.length !== suggestions.length) {
+                    $log.error(
+                      'Number of suggestion threads doesn\'t match number of' +
+                       'suggestion objects');
+                  }
+                  for (var i = 0; i < suggestionThreads.length; i++) {
+                    for (var j = 0; j < suggestions.length; j++) {
+                      if (suggestionThreads[i].thread_id ===
+                          suggestions[j].suggestion_id) {
+                        var suggestionThread = (
+                          SuggestionThreadObjectFactory.createFromBackendDicts(
+                            suggestionThreads[i], suggestions[j]));
+                        $scope.questionSuggestionThreads.push(suggestionThread);
+                        break;
+                      }
+                    }
+                  }
+                });
+            });
+          };
+
+          $scope.setQuestionUnderReview = function(questionSuggestionThread) {
```

Comment:
setActiveQuestion, setActiveSuggestionId, etc.

What happens to suggestionReviewMessage?

Reply:
Done